src/main/java/com/aws/greengrass/detector/config/Config.java

```diff
@@ -7,19 +7,24 @@
 
 import com.aws.greengrass.componentmanager.KernelConfigResolver;
 import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
 import com.aws.greengrass.util.Coerce;
 
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 
 public class Config {
+    private final Logger logger = LogManager.getLogger(Config.class);
+
     static final String INCLUDE_IPV4_LOOPBACK_ADDRESSES_CONFIG_KEY = "includeIPv4LoopbackAddrs";
     static final String INCLUDE_IPV4_LINK_LOCAL_ADDRESSES_CONFIG_KEY = "includeIPv4LinkLocalAddrs";
-    static final int DEFAULT_MQTT_PORT = 8883;
+    static final String DEFAULT_PORT_CONFIG_KEY = "defaultPort";
     static final boolean DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES = false;
     static final boolean DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES = false;
+    static final int DEFAULT_PORT = 8883;
 
-    private AtomicInteger mqttPort = new AtomicInteger(DEFAULT_MQTT_PORT);
+    private AtomicInteger defaultPort = new AtomicInteger(DEFAULT_PORT);
```

Comment:
A side question: why do we use `AtomicXXX` here? Are we using their atomic compound operations? If we just do primitive assignment, I think it's intrinsically atomic in Java.